tests/globalkilltest/global_kill_test.go

```diff
@@ -167,42 +273,29 @@ func (s *TestGlobalKillSuite) stopService(name string, cmd *exec.Cmd, graceful b
 	return nil
 }
 
-func (s *TestGlobalKillSuite) startPDProxy() (proxy *pdProxy, err error) {
-	from := fmt.Sprintf(":%s", *pdProxyPort)
-	if len(s.pdCli.Endpoints()) == 0 {
-		return nil, errors.New("PD no available endpoint")
-	}
-	u, err := url.Parse(s.pdCli.Endpoints()[0]) // use first endpoint, as proxy can accept ONLY one destination.
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	dst := u.Host
-
-	var p pdProxy
-	p.AddRoute(from, to(dst))
-	if err := p.Start(); err != nil {
-		return nil, err
-	}
-	log.Infof("start PD proxy: %s --> %s", from, dst)
-	return &p, nil
-}
-
 func (s *TestGlobalKillSuite) connectTiDB(port int) (db *sql.DB, err error) {
 	addr := fmt.Sprintf("127.0.0.1:%d", port)
 	dsn := fmt.Sprintf("root@(%s)/test", addr)
 	sleepTime := 250 * time.Millisecond
 	startTime := time.Now()
-	for i := 0; i < 5; i++ {
+	maxRetry := 5
+	for i := 0; i < maxRetry; i++ {
 		db, err = sql.Open("mysql", dsn)
 		if err != nil {
 			log.Warnf("open addr %v failed, retry count %d err %v", addr, i, err)
+			if i == maxRetry-1 {
+				return
+			}
 			continue
 		}
 		err = db.Ping()
 		if err == nil {
 			break
 		}
 		log.Warnf("ping addr %v failed, retry count %d err %v", addr, i, err)
+		if i == maxRetry-1 {
```

Comment:
ditto

Reply:
however this one is to make the test more robust, previously, if the connection is not ready after 5 attempts, it silently exits (because the err is occluded by the following `err = db.Close()`, this is to expose the error directly.

Reply:
The error will be Warnf in the previous row line?